gobblin-core/src/main/java/gobblin/publisher/BaseDataPublisher.java

```diff
@@ -133,12 +146,9 @@ public void publishData(Collection<? extends WorkUnitState> states) throws IOExc
           this.fss.get(branchId).mkdirs(publisherOutputDir.getParent());
         }
 
-        if (this.fss.get(branchId).rename(writerOutputDir, publisherOutputDir)) {
-          LOG.info(String.format("Moved %s to %s", writerOutputDir, publisherOutputDir));
-          writerOutputPathsMoved.add(writerOutputDir);
-        } else {
-          throw new IOException("Failed to move from " + writerOutputDir + " to " + publisherOutputDir);
-        }
+        LOG.info(String.format("Moving %s to %s", writerOutputDir, publisherOutputDir));
+        writerOutputPathsMoved.add(writerOutputDir);
```

Comment:
I moved `writerOutputPathsMoved.add(writerOutputDir)` after `parallelRunner.renamePath(writerOutputDir, publisherOutputDir)`, which should have the same semantics as before.